src/actions/functionStore.js

```diff
@@ -0,0 +1,75 @@
+//TODO: I don't like the filename... to many files with the same name
+
+
+//TODO: proxy this through the API?
+const storeUrl = 'https://raw.githubusercontent.com/openfaas/store/master/functions.json';
+
+//TODO memoize? Does this need to be loaded every component load or just ever page load?
+export function loadStoreFunctions() {
+  return dispatch => {
+    dispatch({type:'STORE_LOAD', status:'PENDING'})
+
+    fetch(storeUrl)
+        .then(res => res.json())
+        .then(response => {
+            if (response && response.functions) {
+                dispatch({type:'STORE_LOAD', status:'SUCCESS', storeList: response.functions})
+            }
+        }).catch(error => {
+          console.error(error)
+          dispatch({type:'STORE_LOAD', status:'FAILED', error})
+        })
+  }
+}
+
+
+export function deployStoreFunction(event, func) {
+  return dispatch => {
+
+    dispatch({
+      type: 'FUNCTION_DEPLOY',
+      status: 'PENDING',
+      functionName:func.name
+    })
+
+    //TODO: x86 only?
```

Comment:
Yes, for now. This is the same functionality as is in the current gateway. Perhaps we could query the gateway for the architecture it's deployed on? 
Something for another time...

Reply:
Ok, before resolving, I'll pull this out into a gh issue